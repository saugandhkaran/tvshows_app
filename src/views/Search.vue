<template>
  <div class="search">
    <h2 class="flex-container center">Search about your favourite shows here.</h2>
    <search-show>
      <div slot-scope="{ data: searchResult, loading: loading, error: error, searchShowDetails: searchShowDetails }">
        <div class="flex-container center">
          <div class="form-control">
            <input type="text" v-model="searchterm" placeholder="Search by show name" />
            <button type="submit" class="btn-primary" @click="searchShowDetails(searchterm)">Search</button>
          </div>
        </div>
        <h2 class="flex-container center" v-if="loading">Searching results for '{{searchterm}}'</h2>
        <h2 class="flex-container center" v-if="!loading && !searchResult">No results for '{{searchterm}}'</h2>
        <h2 class="flex-container error center" v-if="error">Something went wrong</h2>
        <div class="flex-container section overflow-x" v-if="searchResult.length">
          <tv-show-card @click.native="goToShowDetail(result.show.id)" v-for="result in searchResult" :key="result.show.id" :showObject="result.show"></tv-show-card>
        </div>
      </div>
    </search-show>
  </div>
</template>

<script>
import TvShowCard from '@/components/TVShowCard.vue'
import SearchShow from '@/components/renderless/SearchShow.vue'
export default {
  name: 'Search',
  components: {
    'tv-show-card': TvShowCard,
    'search-show': SearchShow
  },
  data () {
    return {
      searchterm : ''
    }
  },
  methods: {
    goToShowDetail(id) {
      this.$router.push(`/browse/${id}`);
    }
  }
}
</script>

<style lang="scss" scoped>
h2 {
  padding-left: 20px;
}
</style>
