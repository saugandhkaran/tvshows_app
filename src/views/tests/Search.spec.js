import { shallowMount, createLocalVue } from '@vue/test-utils'
import VueRouter from 'vue-router'
import Search from '../Search';

const localVue = createLocalVue();
localVue.use(VueRouter)
const router = new VueRouter();

describe('Search', () => {
  
  it('should match the Search html', () => {
    const wrapper = shallowMount(Search, {
      stubs: {
        'search-show': 'search-show',
      },
    })
    expect(wrapper.element).toMatchSnapshot();
  });
  
  it('should route to path on click of TV show card', () => {
    const wrapper = shallowMount(Search, {
      localVue,
      router,
      stubs: {
        'search-show': 'search-show',
      },
      $route: {
        path: '/test'
      }
    });
    wrapper.vm.goToShowDetail('1');
    expect(wrapper.vm.$route.path).toEqual('/browse/1')
  });
});
