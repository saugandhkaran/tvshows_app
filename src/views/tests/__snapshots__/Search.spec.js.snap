// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should match the Search html 1`] = `
<div
  class="search"
>
  <h2
    class="flex-container center"
  >
    Search about your favourite shows here.
  </h2>
   
  <div />
</div>
`;
